plugins/besecure.rb

```diff
@@ -0,0 +1,310 @@
+        return ''
+      end
+      
+      if @debug
+        print_status(res)
+      end
+
+      body = ''
+      begin
+        body = JSON.parse(res.body)
+      rescue JSON::ParserError
+        print_error("#{@hostname} - Unable to parse the response")
+        return ''
+      end
+  
+      if body['error']
+        print_error("#{@hostname} - An error occured:")
+        print_error(body)
+        return ''
+      end
+
+      data = body['data']
+      data.each do |item|
+        tbl << [ item['ID'], item['Name'], item['PrettyRange']]
+      end
+
+      # print_good(body)
+  
+      print_good("beSECURE list of reports")
+      print_line
+      print_line tbl.to_s
+      print_line
+    end
+
+    def cmd_besecure_report_download(*args)
+      if args?(args, 4)
+        req = Net::HTTP::Post.new('/json.cgi', initheader={'Host'=>@hostname})
+        format_file = args[1]
+        req.set_form_data({'apikey' => @apikey, 'primary' => 'vulnerabilities', 'secondary' => 'report', 'action' => 'getreport', 'network' => args[0], 'format' => format_file})
+
+        if @debug
+          print_status(req.body)
+        end
+
+        http = Net::HTTP::new(@hostname, 443)
+        http.use_ssl = true
+        res = http.start {|http| http.request(req)}
+
+        unless res
+          print_error("#{@hostname} - Connection timed out")
+          return ''
+        end
+        
+        if @debug
+          print_status(res)
+        end
+
+        body = ''
+        begin
+          body = JSON.parse(res.body)
+        rescue JSON::ParserError
+          print_error("#{@hostname} - Unable to parse the response")
+          return ''
+        end
+    
+        if body['error']
+          print_error("#{@hostname} - An error occured:")
+          print_error(body)
+          return ''
+        end
+
+        # if @debug
+        #   print_status(body)
+        # end
+
+        decompressed = ''
+        if format_file != 'json'
+          compressed_base64 = body["compresseddata"]
+          compressed = Base64.decode64(compressed_base64)
+          decompressed = Zlib::Inflate.inflate(compressed)
+        else
+          decompressed = body
+        end
+
+        if @debug
+          print_status(decompressed)
+        end
+
+        ::FileUtils.mkdir_p(args[2])
+        name = ::File.join(args[2], args[3])
+        print_status("Saving report to #{name}")
+        output = ::File.new(name, "w")
+        output.puts(decompressed)
+        output.close
+      else
+        print_status("Usage: besecure_report_download <network_id> <format_name> <path> <report_name>")
+      end
+    end
+
+    def cmd_besecure_report_import(*args)
+      if args?(args, 2)
+        tempfile = Tempfile.new('results')
+
+        cmd_besecure_report_download(args[0], 'nbe', File.dirname(tempfile) + "/", File.basename(tempfile) )
+
+        print_status("Importing report to database.")
+        framework.db.import_file({:filename => tempfile})
```

Comment:
Not a blocker, but it might be worthwhile detecting if the database is connected or not. It gracefully ignores a database not being connected, so it's not much of an issue issue.

Reply:
How do you check if the db is present? I haven't seen (importers) others do that

Where do you check this? at the beginning of the plugin?

Reply:
You should be able to use `framework.db.active`